pkg/server/node.go

```diff
@@ -858,10 +868,82 @@ func (n *Node) addStore(ctx context.Context, store *kvserver.Store) {
 		// initialization process.
 		log.Fatal(ctx, "attempting to add a store without a version")
 	}
+	store.TODOEngine().RegisterDiskSlowCallback(func(info pebble.DiskSlowInfo) {
+		n.onStoreDiskSlow(ctx, store.StoreID(), info)
+	})
 	n.stores.AddStore(store)
 	n.recorder.AddStore(store)
 }
 
+func (n *Node) onStoreDiskSlow(
+	ctx context.Context, storeID roachpb.StoreID, info pebble.DiskSlowInfo,
+) {
+	// Emit a disk slow event for each slow store that reports a disk slow event.
+	// After diskSlowClearInterval since the last disk slow event, we emit a disk
+	// slowness cleared event. Any other disk slow events for the same store in
+	// between are ignored as duplicates.
+	//
+	// TODO(bilal): Store stats on events observed and coalesced events.
+	const diskSlowClearInterval = 30 * time.Second
```

Comment:
Do we want to make this configurable? I imagine there will be differing interpretations of what it means for slowness to clear.

Reply:
We already have maxSyncDuration that's configurable, and we can make this a multiple of that (or just equal to that). What do you think about that idea?

Reply:
> or just equal to that

This seems reasonable. If the stall is longer than this, we'll crash, right?

Reply:
Not necessarily. If we're seeing write events that are consistently slow at about 5s or so per request, we'll see a prolonged disk slowness event in the event log, but no crash because no individual write took >20s.

But you're right, that more commonly we either see slowness events around 5s that go away quickly, or >20s in which case there's just a crash. So I made the `maxSyncDuration` change.